### PR TITLE
Add OWASP Zap API scanner to CI/CD

### PR DESCRIPTION
**Pull request by adamkorynta**
## Problem Description

SonarCloud only provides a static analysis of security vulnerabilities. A runtime analysis is also needed.

## Solution

Start docker containers with the Postgres database and jetty installation and run the OWASP Zap automated tool from another docker container against the OpenDCS API.

## how you tested the change

Testing docker containers locally and from GitHub Actions. 
The GitHub summary shows the summary of the Markdown output report. The HTML report is attached as a build artifact.

## Where the following done:

- [X] Was relevant documentation updated?
- [X] Were relevant config element (e.g. XML data) updated as appropriate

### CODE REVIEW

File: gradle.properties
Comment (MikeNeilson) [inline gradle.properties:1]: this should be a `gradle.properties.example` with `gradle.properties` excluded from the build.
Less risk of credential spillage that way.

File: odcsapi-jetty-standalone/build.gradle
Comment (MikeNeilson) [inline odcsapi-jetty-standalone/build.gradle:67]: FYI opendcs 7.0.5+ have support for retrieving credentials from the environment

use AuthSource.getCredentials(<the db auth file string>);

See https://github.com/opendcs/opendcs/blob/2fc798892dc1d49840e579e75ff0ca43bc72c54d/src/test/java/org/opendcs/authentication/EnvironmentAuthSourceTest.java#L27 for a full examples

File: odcsapi-jetty-standalone/config/user.properties
Comment (MikeNeilson) [inline odcsapi-jetty-standalone/config/user.properties:48]: this line can be changed to the EnvAuth spec as in the example above.

**Comment by MikeNeilson** [comment]
FYI, I'm working in this PR specifically for this project: https://github.com/opendcs/opendcs/pull/387

You might want to hold off on some of this. The above project (which already exists, that PR is just making usable artifacts) already handles database initialization and initial dataload.

It's not unreasonable at this stage to inform the dev that they need an existing database setup and we're working on some automation.

**Comment by adamkorynta** [comment]
Currently the initial database table load isn't working right and the scripts need a couple of hacks to get the schema to install correctly. I'll hold off on anymore improvement to that until after your opendcs PR is merged.

Additionally, the OWASP Zap docker scan is occurring locally, but the swagger.json isn't auto-generated so the "odcsapi" endpoint winds up being relative to the swagger.json endpoint (resulting in scans of funny endpoints like `"http://opendcs:8000/swagger.json/odcsapi/reflists"`) so I will have to pause on this effort until we can get the swagger.json autogenerated.

**Comment by MikeNeilson** [comment]
This PR: https://github.com/opendcs/opendcs/pull/433 will probably be more appropriate.
I'm still working out issues with data in classpaths and such (trying to keep it clean to the downstream programs)

But this will certainly make the actual schema install easier.

File: odcsapi-jetty-standalone/build.gradle
Comment (adamkorynta) [reply odcsapi-jetty-standalone/build.gradle]: Perfect, thanks!

**Review by wjonassen — APPROVED** [review]

File: opendcs-rest-api/src/main/resources/swaggerui/open_api.json
Comment (adamkorynta) [reply opendcs-rest-api/src/main/resources/swaggerui/open_api.json:10]: yeah, I believe the "servers" section should be completely removed for that reason, I just don't know how it automatically gets picked up.

Comment (adamkorynta) [reply opendcs-rest-api/src/main/resources/swaggerui/open_api.json:10]: I added a comment to: https://github.com/opendcs/rest_api/issues/78

**Pull request merged**
